### ISSUE TITLE
It takes over one second to source nvm.fish

### ISSUE THREAD

**Issue by holyjak** [passcod/nvm-fish-wrapper#9]
Hello,

on my new, high-end Mac with Yosemite it takes over 1 sec to source `nvm.fish`. Is that normal? If not, do you have any troubleshooting tips? (PS: I have installed nvm via brew)

Thank you for your great work!

Best Regards, Jakub Holy

**Comment by passcod** [comment]
Hi, as noted on the README, OS X is unsupported (i.e. it seems to work, with numerous caveats) (as I do not have a Mac, and nobody so far has stepped up to the task), so I'm not able to do much on that issue.

No, nvm.fish shouldn't take more that a few (hundred, at worst) milliseconds to load. That should be the case regardless of how "high-end" your computer is — unless you're running on a computer from the last decade or so, maybe.

As nvm.fish is a wrapper around nvm, you could check:
- the time it takes to source nvm in bash,
- comment out the last line of nvm.fish and try again,
- install a node version, verify or set the `default` alias, and try again (with the last line as above _un_commented),
- actually time (using a stopwatch or something on the command-line — keeping in mind that the overhead of launching a program may be higher than the actual time you're measuring) the sourcing of nvm.fish to make sure it's not bias or subjective time passing (have you ever observed that phenomenon where a single second _feels_ like much longer?)

In the end, though, you're probably in the same boat as the people in #7 and #8 — this wrapper is woefully untested on OS X, and compatibility is tricky to get right… especially without help.

**Comment by alex7kom** [comment]
I'm using OS X Yosemite and so far `nvm-fish-wrapper` works flawlessly for me. I'd like to help but I can't reproduce any of the current OS X issues.
If it can be of any help I'm using fish 2.1.2 with the latest Xcode command tools and the latest nvm (0.24.1) installed the manual way (I use MacPorts, not Homebrew).

**Comment by passcod** [comment]
> (I use MacPorts, not Homebrew)

Oh damn that could be something. #7 indicated an issue to do with the `manpath` utility… and in both those cases I'm pretty sure that those systems are using Homebrew's `coreutils`. That may make a difference.

**Comment by holyjak** [comment]
Hm, I have discovered that 'brew --prefix', which I suppose should take "no
tine" takes over 200ms so it's well possible something is wrong with
brew/coreutils .

## 

Forget software. Strive to make an impact, deliver a valuable change.

(Vær så snill og hjelp meg med å forbedre norsken min – skriftlig og
muntlig. Takk!)

Jakub Holy
Solutions Engineer | +47 966 23 666
Iterate AS | www.iterate.no
The Lean Software Development Consultancy
- http://theholyjava.wordpress.com/ -
- apr. 2015 12:19 skrev "Félix Saparelli" notifications@github.com:

**Comment by passcod** [comment]
Homebrew is written in Ruby, which takes some startup time. Try running `ruby -e "puts :Hello"` and see if the time that takes is similar. Also try running some other non-trivial non-Homebrew and see what time those take, as comparison.

**Comment by holyjak** [comment]
So it is normal that `brew --prefix <something>` takes a while - e.g. [mentioned here](http://kleinmann.org/2015/01/18/decreasing-my-zsh-startup-time/).

BTW I believe I have used nvm-wrapper on my previous Mac that also had Yosemite (likely a somewhat different version of both OS and command0line tools for XCode) and it worked just fine. 

Regarding `coreutils` - I have installed those via brew so perhaps that could be the culprit.

@passcod Thanks for your troubleshooting tips. I will try that and if I find something useful, share it.

**Comment by passcod** [comment]
Yea, that's normal. (Regarding that article… back when I did have a Mac, I think Homebrew dumped a bunch of environment variables in your shell, (you can check your env with `env`) and that included the `BREW_PREFIX` or similar, which would be much better for speeding up shells and scripts than hard-coding the `/usr/local/opt/whatever` path. Maybe these vars are still there! Anyway, not relevant :P)

**Comment by holyjak** [comment]
**FIXED** by removing the code that I have added to `nvm.fish` (sorry for forgetting about it!):

```
if test -f (brew --prefix nvm)/nvm.sh
   set -x NVMSH (brew --prefix nvm)/nvm.sh
else
  set -x NVMSH ~/.nvm/nvm.sh
end
```

and re-adding `set -x NVMSH ~/.nvm/nvm.sh`and fixing the missing `~/.nvm/nvm.sh` it was meant to solve by symlinking `ln -s /usr/local/opt/nvm/nvm.sh ~/.nvm/nvm.sh`. 

Other observarions:
1. If I do not set `NVMSH`, `. nvm.fish` takes ~30ms but (understandably) bash will complain about missing file name in calls to `source`; however, `node` and `npm` are on the path and even if `nvm` commands do not work, I can still use bash nvm when I need them
2. If `NVMSH` is set properly, `nvm` in fish is fully functional but sourcing it takes over 700ms (`time fish -l ~/.config/fish/nvm-wrapper/nvm.fish` after having removed my `config.fish` to get a bare-bone fish :))

Thank you and sorry for wasting your time!

PS: I used bash' nvm.sh to install a node version.
PPS: `brew --prefix nvm` itself takes 0.4s; also, using the `../opt/../nvm.sh` sets different env vars (as described in https://github.com/passcod/nvm-fish-wrapper/issues/8#issuecomment-94318495) which likely causes other problems
